src/metatrain/utils/additive/_base_composition.py

```diff
@@ -316,12 +320,10 @@ def forward(
         dtype = systems[0].positions.dtype
         self._sync_device_dtype(device, dtype)
 
-        system_indices, sample_labels_per_atom = _get_system_indices_and_labels(
-            systems, device
-        )
+        _, sample_labels_per_atom = _get_system_indices_and_labels(systems, device)
 
         predictions: Dict[str, TensorMap] = {}
-        for output_name, model_output in outputs.items():
+        for output_name in outputs.keys():
```

Comment:
No need for the `keys()` here

```suggestion
        for output_name in outputs:
```